src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -264,11 +265,15 @@ private void UpdateReferenceLocationsTask(Task<IInlineRenameLocationSet> allRena
 
             var asyncToken = _asyncListener.BeginAsyncOperation("UpdateReferencesTask");
             _allRenameLocationsTask = allRenameLocationsTask;
-            allRenameLocationsTask.SafeContinueWith(
-                t => RaiseSessionSpansUpdated(t.Result.Locations.ToImmutableArray()),
+            allRenameLocationsTask.SafeContinueWithFromAsync(
+                async t =>
+                {
+                    await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_cancellationTokenSource.Token);
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.